src/interfaces/earnings.ts

```diff
@@ -47,16 +52,16 @@ export class EarningsInterface<C extends ChainId> extends ServiceInterface<C> {
   }
 
   async assetEarnings(assetAddress: Address): Promise<AssetEarnings> {
-    const response = (await this.yearn.services.subgraph.fetchQuery(VAULT_EARNINGS, {
+    const response = await this.yearn.services.subgraph.fetchQuery<VaultEarningsResponse>(VAULT_EARNINGS, {
       vault: assetAddress
-    })) as VaultEarningsResponse;
-
-    const vault = response.data.vault;
+    });
 
-    if (!vault) {
+    if (!response?.data || !response.data?.vault) {
```

Comment:
Thought it might be safer to check for `response.data` first, and if we don't have it then throw